### PR TITLE
bring rna branch up to date

### PR DESCRIPTION
**Pull request by saupchurch**
PR to bring server/rna up to date with latest development version of rna api code

### CODE REVIEW

**Comment by jeromekelleher** [comment]
Looks good @saupchurch. Perhaps we should just bring the rna branch up to date with develop separately, and keep these changes out of the PR? @dcolligan, can you check to see if the rna branch can be fast-forwarded, and if so, make the changes please?

**Comment by dcolligan** [comment]
Just merged develop into rna

**Comment by saupchurch** [comment]
updated _protocol_definitions.py using process_schema.py build should now pass

File: tests/end_to_end/test_oidc.py [hunk 1]
```diff
@@ -8,9 +8,10 @@
 
 import requests
 import subprocess
-from lxml import html
 from urlparse import urlparse
 
+import lxml.html as html
+
```
Comment (macieksmuga) [inline tests/end_to_end/test_oidc.py]: Is there a new python style guideline I'm missing? The `from <module> import <submodule>` seems much less stilted than `import <module>.<submodule> as <submodule>`, with exactly the same end result.

**Pull request merged**